src/main/java/teammates/ui/controller/InstructorCoursesPageData.java

```diff
@@ -1,35 +1,192 @@
+                if (!curInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_COURSE)) {
+                    deleteButton.setAttribute("disabled", "disabled");
+                }
+                
+                actionsParam.add(unarchivedButton);
+                actionsParam.add(deleteButton);
+                
+                ArchivedCoursesTableRow row = new ArchivedCoursesTableRow(Sanitizer.sanitizeForHtml(course.id), 
+                                                                          Sanitizer.sanitizeForHtml(course.name), actionsParam);
+                archivedCourses.getRows().add(row);
+            }
+        }
+        
+        return archivedCourses;
+    }
     
-    /* Values to show in the form fields (in case reloading the page after a 
-     *   failed attempt to create a course)*/
-    public String courseIdToShow;
-    public String courseNameToShow;
+    private ActiveCoursesTable extractActiveCourses(List<CourseDetailsBundle> courseBundles) {
+        ActiveCoursesTable activeCourses = new ActiveCoursesTable();
+        Logic logic = new Logic();
+        
+        CourseDetailsBundle.sortDetailedCoursesByCourseId(courseBundles);
+        
+        int idx = -1;
+        for (CourseDetailsBundle courseBundle : courseBundles) {
+            CourseAttributes course = courseBundle.course;
 
+            InstructorAttributes curInstructor = logic.getInstructorForGoogleId(course.id, account.googleId);
+
+            if (!Logic.isCourseArchived(course.id, curInstructor.googleId)) {
+                idx++;
+                
+                List<ElementTag> actionsParam = new ArrayList<ElementTag>();
+                
+                ElementTag enrollButton = new ElementTag();
+                enrollButton.setContent("Enroll");
+                enrollButton.setAttribute("class", "btn btn-default btn-xs t_course_enroll" + idx);
+                enrollButton.setAttribute("href", getInstructorCourseEnrollLink(courseBundle.course.id));
+                enrollButton.setAttribute("data-toggle", "tooltip");
+                enrollButton.setAttribute("data-placement", "top");
+                enrollButton.setAttribute("title", Const.Tooltips.COURSE_ENROLL);
+                if (!curInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT)) {
+                    enrollButton.setAttribute("disabled", "disabled");
+                }
+                
+                ElementTag viewButton = new ElementTag();
+                viewButton.setContent("View");
+                viewButton.setAttribute("class", "btn btn-default btn-xs t_course_view" + idx);
+                viewButton.setAttribute("href", getInstructorCourseDetailsLink(courseBundle.course.id));
+                viewButton.setAttribute("data-toggle", "tooltip");
+                viewButton.setAttribute("data-placement", "top");
+                viewButton.setAttribute("title", Const.Tooltips.COURSE_DETAILS);
```

Comment:
It may be good to have a convenience method instead of duplicating this whole chunk for each button considering they share the same types of attributes.